airflow/models/taskinstance.py

```diff
@@ -787,6 +788,38 @@ def error(self, session: Session = NEW_SESSION) -> None:
         session.merge(self)
         session.commit()
 
+    @classmethod
+    @internal_api_call
+    @provide_session
+    def get_task_instance(
+        cls,
+        dag_id: str,
+        run_id: str,
+        task_id: str,
+        map_index: int,
+        select_columns: bool = False,
+        lock_for_update: bool = False,
+        session: Session = NEW_SESSION,
+    ) -> TaskInstance | None:
+        query = (
+            session.query(*TaskInstance.__table__.columns) if select_columns else session.query(TaskInstance)
+        )
+        query = query.filter_by(
+            dag_id=dag_id,
+            run_id=run_id,
+            task_id=task_id,
+            map_index=map_index,
+        )
+
+        if lock_for_update:
```

Comment:
how does it behave in Internal API? 
Does it release the rows when session is destroyed (I expect it) - so just after the results are returned

We may need to think about the scenario when the row-locking is important for potential race conditions or consistency, but I see it's outside of this PR (as we don't use it)

Reply:
I think a good idea would be to split that to two methods. We definitely do not want to have a case where we have lock_for_update=True over the internal API so rather than just leave it here, I think we could probably duplicate that method (with some DRY-internal method query) into:

```


# no internal api 
@provide_session
def get_task_instance_with_lock(
        cls,
        dag_id: str,
        run_id: str,
        task_id: str,
        map_index: int,
        select_columns: bool = False,
        session: Session = NEW_SESSION)
    _get_task_instance(...., lock_for_update=True)



@provide_session
@internal_api_call
def get_task_instance_no_lock(
        cls,
        dag_id: str,
        run_id: str,
        task_id: str,
        map_index: int,
        select_columns: bool = False,
        session: Session = NEW_SESSION):
    _get_task_instance(...., lock_for_update=False)

def _get_task_instance(...

```